scripts/airports.ts

```diff
@@ -0,0 +1,165 @@
+
+  if (airportIndex !== -1) {
+    return airportName.slice(0, airportIndex) + "Flughafen";
+  }
+
+  return airportName + " Flughafen";
+}
+
+function translateAirportCity(airport: AiportDataSource): string {
+  const rows = airportsGermanCitiesContent.split("\n");
+  let city = airport.city;
+
+  rows.forEach((row) => {
+    const [iata, , cityTranslation] = row.split(",");
+    if (airport.iata === iata) {
+      city = cityTranslation;
+    }
+  });
+
+  return city;
+}
+
+function filteredLargeMediumAirports(airports: AiportDataSource[]): Aiport[] {
+  return airports
+    .filter(
+      (airport) =>
+        (airport.type === "large_airport" ||
+          airport.type === "medium_airport") &&
+        airport.scheduled_service === "yes",
+    )
+    .map((airport) => {
+      return {
+        iata: airport.iata,
+        country_code: airport.country_code,
+        country:
+          countriesTranslation.getName(airport.country_code, germanLocale) ??
+          "",
+        airport: translateAirportName(airport.airport),
+        latitude: airport.latitude,
+        longitude: airport.longitude,
+        city: translateAirportCity(airport),
+      };
+    });
+}
+
+function parseAiportDataSource(content: string): AiportDataSource[] {
+  const rows = content.split("\n");
+  const airports: AiportDataSource[] = [];
+  rows.slice(1).forEach((row) => {
+    const [
+      ,
+      ,
+      type,
+      name,
+      latitude,
+      longitude,
+      ,
+      ,
+      country_codeData,
+      ,
+      city,
+      scheduled_service,
+      ,
+      iataData,
+    ] = row.split(",");
+
+    const airportData = {
+      iata: iataData,
+      country_code: country_codeData,
+      airport: name,
+      latitude: latitude,
+      longitude: longitude,
+      city: city,
+      type: type,
+      scheduled_service: scheduled_service,
+    };
+
+    const airportResult = AirportDataSourceSchema.safeParse(airportData);
+
+    if (airportResult.success) {
+      airports.push(airportResult.data);
+    }
+  });
+
+  return airports;
+}
+
+async function fetchAllAirports(): Promise<Aiport[]> {
+  const res = await fetch(AIPORTS_URL_DATA_SOURCE, {
+    headers: {
+      "content-type": "text/csv;charset=UTF-8",
+    },
+  });
+
+  if (res.status === 200) {
+    const content = await res.text();
+    const airportsDataSource = parseAiportDataSource(content);
+    return filteredLargeMediumAirports(airportsDataSource);
+  }
+
+  console.log(`Error code ${res.status}. Status text: ${res.statusText}`);
+  return [];
+}
+
+async function fetchAndSaveAirports() {
+  const airports = await fetchAllAirports();
+  const data = JSON.stringify(_.uniqBy(airports, "iata"));
+  const filePath = "data/airports/data.json";
+
+  fs.writeFile(filePath, data, (error) => {
+    if (error) {
+      console.error(error);
+      throw error;
+    }
+  });
+  console.log(`${filePath} written correctly`);
+}
```

Comment:
I believe the comment about writing to the file synchronously is referring to [this](https://github.com/digitalservicebund/a2j-rechtsantragstelle/pull/927/files#diff-5e57d179fb325d62c54ca32067b8eddc9c34a1491049541732f532a75572afa2R11-R13) part? And if we could also write to the file using `writeFileSync`: 

```suggestion
  fs.writeFileSync(filePath, data);
  console.log(`${filePath} written correctly`);
}
```

Reply:
Ohh I got what @pgurusinga meant now. It sounds much better like this, I'll change it!

Reply:
sehrrrrrr schöööööön